ConcordiumWallet/Views/AccountsView/SendFunds/SendFund/SendFundPresenter.swift

```diff
@@ -288,13 +289,15 @@ class SendFundPresenter: SendFundPresenterProtocol {
             .store(in: &cancellables)
 
       
-        Publishers.CombineLatest3(
+        Publishers.CombineLatest4(
             viewModel.$recipientAddress,
             viewModel.$feeMessage,
-            viewModel.$enteredAmount
+            viewModel.$enteredAmount,
+            $isUpdatingTransferCost
         )
         .receive(on: DispatchQueue.main)
-        .map { [weak self] recipientAddress, feeMessage, amount in
+        .map { [weak self] recipientAddress, feeMessage, amount, isUpdatingTransferCost in
+            guard isUpdatingTransferCost == false else { return false }
```

Comment:
Can you please explain what exact problem this is solving? To me it just looks like it could prevent estimation to happen if I input something while the request is being slow. Meaning that this input won't be taken into account in the estimation.

Reply:
previous was done without blocking send button, and there was case when you enter recepinet address, and type amount and press send button.  in this case tx update haven't finished yet, and tx was sent with wrong amount of energy,  much lower than needed.   which cause send failed: insificient energy

Reply:
Ok thanks. What I meant was: what problem does this exact line solve?

You also mentioned blocking the send button. Where does that happen?

Reply:
solving: outdated tx cost data, prevent sending tx while fetching new tx estimation data by disabling send button. 
where: basically here 
``` CombineLatest4(
            viewModel.$recipientAddress,
            viewModel.$feeMessage,
            viewModel.$enteredAmount,
            $isUpdatingTransferCost
        )
``` 
, this is 4 publishers that are somehow disables send button, in out case by checking  `isUpdatingTransferCost `

Reply:
My bad, I misunderstood the context of the code. It would be helpful with a comment above the pipeline explaining that it's for updating the enabledness state of the button.

I still have one minor suggestion though ;)

```suggestion
            if isUpdatingTransferCost { return false }
```